### ISSUE TITLE
ENOTSUP on Thumbs.db file

### ISSUE THREAD

**Issue by paulovictorv** [isaacs/node-glob#208]
I'm getting ENOTSUP when glob tries to read a Thumbs.db as a folder (I think). The drive is a Windos network drive.

First example:
``Z:\\DPT1\\CLIP\\**\\' + qry + '\\CORRECTED\\**\\*.pdf`` breaks when ``Thumbs.db`` can be found below CORRECTED directory.

Second example:
``Z:\\DPT1\\CLIP\\**\\' + qry + '\\CORRECTED\\*.pdf`` does not break even if ``Thumbs.db`` exists.

Not really sure if this is a bug with glob, but posting here at least for awareness.

**Comment by isaacs** [comment]
What version of glob are you using?  There was a very recent change to how ENOTSUP is handled.

**Comment by paulovictorv** [comment]
Hi, I'm using the latest release. 

``"glob": "5.0.13"``

**Comment by isaacs** [comment]
Weird... I just added a test that exercises that bit of the code, and it works fine.

Can you please provide a full reproducible test case, and the exact error you see when running it?

**Comment by paulovictorv** [comment]
Sure.

My folder structure is like this:
``Z:\DPT1\CLIP\DPT1-CLIP-01\CORRECTED\**\*.pdf``
The part where I have the double stars can contain folders/files. Thumbs.db happens to occur on the double star part of the path.

This is the error message as printed by the code.

    { [Error: ENOTSUP, scandir 'Z:\DPT1\CLIP\DPT1\CLIP\DPT1-CLIP-01\CORRECTED\Thumbs.db']
    errno: -4049,
    code: 'ENOTSUP',
    path: 'Z:\\DPT1\\CLIP\\DPT1\\CLIP\\DPT1-CLIP-01\\CORRECTED\\Thumbs.db' }

As a reminder, ``Z`` is a network drive.

Let me know if you need more information! Thanks

**Comment by isaacs** [comment]
That's strange.  Getting an `ENOTSUP` on a readdir call should be exactly the same as getting `ENOTDIR`.

Are you quite sure that it's `glob@5.0.13` that's reporting this?  Can you run `npm ls glob` to see if there are any other versions in use?

**Comment by paulovictorv** [comment]
It reported ``glob@5.0.13`` sadly :/

Just to be certain, I cleaned my node_modules dir, ``npm install`` it, but still got the same error.

Also, I was doing some tests, and glob seems to fail with some files (in this case it was a pdf). So the error might not be tied only to  ``Thumbs.db``. The error code is the same as before mentioned.

**Comment by paulovictorv** [comment]
Well, I've managed to get around this error by using ``{strict: false}`` as an option, after looking at the source code and seeing the guard around the ``abort``.

It's still weird that this error happens anyways.

**Comment by isaacs** [comment]
Ohhh... are you using `glob.sync` or `glob`?

**Comment by paulovictorv** [comment]
I'm using glob, as the examples in the readme shows. I don't know which one I'm using

**Comment by isaacs** [comment]
Oh, sorry, I'm dumb, apparently.

```
* c47d451 isaacs (HEAD -> master, tag: v5.0.14, origin/master) v5.0.14 (12 seconds ago)
* 821fac8 isaacs Handle ENOTSUP for sync glob as well as async (4 minutes ago)
* 9625618 isaacs Test for when readdir raises ENOTSUP (25 hours ago)
* 0a2b519 isaacs Generate fixtures more effectively, with -O instead of eval (9 days ago)
* f96190b isaacs Use js for benchmark cleanup (12 days ago)
* 957fd93 isaacs Fix some 'use strict' errors (12 days ago)
* bf3381e isaacs Treat ENOTSUP like ENOTDIR in readdir (12 days ago)
* 507733d isaacs (tag: v5.0.13) v5.0.13 (12 days ago)
```

Notice anything interesting?  I never released the ENOTSUP fix.  That explains it!

The good news is that I never released *half* the ENOTSUP fix, so it now works on 5.0.14 for both sync and async glob.  Can you re-install and let me know?

Sorry for not catching that!

**Comment by paulovictorv** [comment]
Hurray! Now it works, thanks for the help man :)
